modules/ethereum/src/finality.rs

```diff
@@ -214,106 +214,110 @@ pub(crate) fn ancestry<'a, S: Storage>(
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use crate::tests::{genesis, validator, validators_addresses, InMemoryStorage};
-	use crate::HeaderToImport;
+	use crate::mock::{custom_test_ext, genesis, validator, validators_addresses, TestRuntime};
+	use crate::{BridgeStorage, HeaderToImport};
 
 	#[test]
 	fn verifies_header_author() {
-		assert_eq!(
-			finalize_blocks(
-				&InMemoryStorage::new(genesis(), validators_addresses(5)),
-				&Default::default(),
-				(&Default::default(), &[]),
-				&Default::default(),
-				None,
-				&Header::default(),
-				0,
-			),
-			Err(Error::NotValidator),
-		);
+		custom_test_ext(genesis(), validators_addresses(5)).execute_with(|| {
+			assert_eq!(
+				finalize_blocks(
+					&BridgeStorage::<TestRuntime>::new(),
+					&Default::default(),
+					(&Default::default(), &[]),
+					&Default::default(),
+					None,
+					&Header::default(),
+					0,
+				),
+				Err(Error::NotValidator),
+			);
+		});
 	}
 
 	#[test]
 	fn prepares_votes() {
```

Comment:
I'd like to see all the stuff within `custom_test_ext()` be cleaned up with some functions, so that the test looks something like this:

```rust
	#[test]
	fn validators_finalize_block_when_threshold_is_met() {
		...

		let header1 = get_header(...);
		let finalized_blocks = get_finalized_blocks(header1, ...);
		assert_eq!(finalized_blocks, Ok(Vec::new()));

		let header2 = get_header(...);
		let finalized_blocks = get_finalized_blocks(header2, ...);
		assert_eq!(finalized_blocks, Ok(Vec::new()));

		let header3 = get_header(...);
		let finalized_blocks = get_finalized_blocks(header3, ...);
		assert_eq!(finalized_blocks, Ok(vec![(1, header1.hash, None)]);
	}

```

I found it hard to follow the meat of what the test was checking with all the setup involved.

Reply:
Logged #111